### PR TITLE
Added Add to Homescreen Button

### PR DESCRIPTION
**Pull request by HiddenTr0ll**
see issue: #592

### Visible/Frontend Changes
- [x] Added an "Add to Homescreen" Button

### CODE REVIEW

**Comment by HiddenTr0ll** [comment]
BeforeInstallPromptEvent is only compatible with Chome (Desktop), Edge, Chrome (Mobile) and Samsung Internet (Mobile) (see https://developer.mozilla.org/en-US/docs/Web/API/BeforeInstallPromptEvent#browser_compatibility) 

TODO
[] Add an alternative, visual guide for users using not supported browser

**Review by sleidig — COMMENTED** [review]
Cool! :smiley: 

The code looks really nice and clean, but maybe you can move it to its own angular component so we can move it around more easily if necessary (and encapsulate code). Don't hesitate to ask if you need some pointers regarding Angular for that. (The cli  `ng generate component` should get you quite far).

However, somehow I don't see the install button on the test deployment :thinking:   Maybe because I have/had the app installed before. Can someone else test and confirm the button shows up?

Do you know what happens if the user dismissed the "install app" prompt on the first visit - will we still get the event on the user's later visits?

File: src/app/core/navigation/navigation/navigation.component.ts [hunk 1]
```diff
@@ -72,4 +76,30 @@ export class NavigationComponent {
...
+  public addToHomeScreen() {
+    this.deferredPrompt.prompt();
+    this.deferredPrompt.userChoice.then((choiceResult) => {
+      if (choiceResult.outcome === "accepted") {
+        console.log("User accepted the A2HS prompt");
+        this.showInstallButton = false;
+      } else {
+        console.log("User dismissed the A2HS prompt");
+      }
+      this.deferredPrompt = null;
```
Comment (TheSlimvReal) [inline src/app/core/navigation/navigation/navigation.component.ts:102]: If the user denied the prompt the first time (`else` case), can the deferred prompt be called again to show the prompt a second time?

Comment (HiddenTr0ll) [reply src/app/core/navigation/navigation/navigation.component.ts:102]: Yes, the button still functions after the installation has been declined.

**Comment by HiddenTr0ll** [comment]
I am also encountering some issues on the test deployment, but I am noticing a pattern. On the live website [https://demo.aam-digital.com/](url) I see the warning `Site cannot be installed: Page does not work offline. The page will not be regarded as installable after Chrome 93, stable release August 2021.`, which warns about an upcomming change to the offline detection, comming in August 2021. When that warning is shown in the console, no `beforeinstallprompt` is fired and therefore no button is shown. On my local testsetup I get this warning about 50% of the time when I refresh the site. But on the occasions where the warning is missing, everything works. Coupled with this warning is a failed fetch from `ngsw-worker.js `(http://puu.sh/Hhlhw/f72c8e702f.png) - this only fails when the warning is there or vice versa. Do you have any ideas?

**Comment by sleidig** [comment]
> I am also encountering some issues on the test deployment, but I am noticing a pattern. On the live website [https://demo.aam-digital.com/](url) I see the warning `Site cannot be installed: Page does not work offline. The page will not be regarded as installable after Chrome 93, stable release August 2021.`, which warns about an upcomming change to the offline detection, comming in August 2021. When that warning is shown in the console, no `beforeinstallprompt` is fired and therefore no button is shown. On my local testsetup I get this warning about 50% of the time when I refresh the site. But on the occasions where the warning is missing, everything works. Coupled with this warning is a failed fetch from `ngsw-worker.js `(http://puu.sh/Hhlhw/f72c8e702f.png) - this only fails when the warning is there or vice versa. Do you have any ideas?

We should definitely follow up on this. Unfortunately I cannot reproduce that warning or error when I try from my machine. Do you have any idea under which exact circumstances this happens? This is on demo.aam-digital.com with app version (visible bottom left after login) 2.27.1? In Chrome Browser (which version? which OS?)?

Maybe updating the angular service-worker package helps. I was planning to do an Angular upgrade the next days.

**Comment by sleidig** [comment]
closing this as it has been dormant for a very long time. I am pointing the issue back to this PR in case a new developer wants to refer back to the work and discussion here.

**Pull request closed without merging**